src/generic_core/social.ts

```diff
@@ -626,6 +626,32 @@ export function notifyUI(networkName :string, userId :string) {
       });
     }
 
+    public processCloudNetworkData(networkData :any) :string {
+      if ((typeof networkData) === 'string') {
+        networkData = JSON.parse(networkData);
+      }
+
+      // If we are enforcing validity, we require that the proxy server be
+      // in our list of approved servers.  If it's not, trigger an error
+      if (globals.settings.enforceProxyServerValidity &&
+          !(networkData.host in globals.settings.validProxyServers)) {
+        throw new Error('Domain policy prohibits unknown proxy servers');
+      }
+
+      // if we have a host key saved, make sure there is no conflict with
+      // the invitation, then used the saved version
+      if (networkData.host in globals.settings.validProxyServers) {
+        if ('hostKey' in networkData &&
+            networkData.hostKey !== globals.settings.validProxyServers[networkData.host]) {
+          throw new Error('Conflict between saved key in settings and key in invitation');
+        }
+
+        networkData.hostKey = globals.settings.validProxyServers[networkData.host];
+      }
+
+      return JSON.stringify(networkData);
```

Comment:
Again, this is weird - suggest this function return `void`, instead modifying its argument on success. What do you think?